pkg/util/provider/machinecontroller/machine.go

```diff
@@ -415,6 +431,28 @@ func (c *controller) getMachineFromNode(nodeName string) (*v1alpha1.Machine, err
 	return machines[0], nil
 }
 
+func inPlaceUpdateLabelsChanged(oldNode, node *corev1.Node) bool {
+	if oldNode == nil || node == nil {
+		return false
+	}
+
+	labelKeys := []string{
+		v1alpha1.LabelKeyNodeCandidateForUpdate,
+		v1alpha1.LabelKeyNodeSelectedForUpdate,
+		v1alpha1.LabelKeyNodeUpdateResult,
+	}
+
+	for _, key := range labelKeys {
+		oldVal, oldOk := oldNode.Labels[key]
+		newVal, newOk := node.Labels[key]
+		if (!oldOk && newOk) || (key == v1alpha1.LabelKeyNodeUpdateResult && oldVal != newVal) {
```

Comment:
For manual in-place update, `selected-for-update` label value on the node can potentially be changed by the user right? In that case shouldn't this check for the change in value for that label as well?
```suggestion
		if (!oldOk && newOk) || (key != v1alpha1.LabelKeyNodeCandidateForUpdate && oldVal != newVal) {
```

Reply:
The `selected-for-update` label when present denotes selection, its value in inconsequential. So this comment is invalid.